src/confu/types.py

```diff
@@ -0,0 +1,43 @@
+"""
+Custom data types to help with env variables
+
+Import from this file (confu.types)
+"""
+import re
+
+
+class TimeDuration(float):
+    """
+    takes in and converts a string to float of seconds.
+        .5 = 500ms
+        90 = 1m30s
+    units accepted are d(days), h(hours), m(minutes),
+    s(seconds) and ms(milliseconds).
+
+    **Arguments**
+
+    - val (`str`)
+    """
+
+    def __new__(cls, value, **kwargs):
+        re_intv = re.compile(r"([\d\.]+)([a-zA-Z]+)")
+        re_validate = re.compile(r"(([\d\.]+)([s,m,ms,h,d]+))*")
+        formatted_val = value.replace(" ", "")
+
+        total = 0.0
+        if not re.fullmatch(re_validate, formatted_val):
+            raise ValueError("unknown unit or format in interval string '%s'" % value)
```

Comment:
@jason20c lets do the more modern `f` formatting here

e.g., should be `f"unknown unit or format in interval string '{value}'"`